lib/Interpreter/InterOp.cpp

```diff
@@ -2165,5 +2165,32 @@ namespace InterOp {
 
     return names;
   }
+
+  std::string GetFunctionArgDefault(TCppFunction_t func, TCppIndex_t param_index)
+  {
+    auto *D = (clang::Decl *)func;
+    auto *FD = llvm::dyn_cast_or_null<clang::FunctionDecl>(D);
+    auto PI = FD->getParamDecl(param_index);
+
+    if (PI->hasDefaultArg())
+    {
+      std::string Result;
+      llvm::raw_string_ostream OS(Result);
+      Expr *DefaultArgExpr = const_cast<Expr *>(PI->getDefaultArg());
+      DefaultArgExpr->printPretty(OS, nullptr, PrintingPolicy(LangOptions()));
```

Comment:
What's in `OS` after this line? Why we need extra work for the floating point case?

Reply:
- Here OS will store default value of arguments 
- Only reason to add extr work for floating point case is same as I mentioned above that when default value was given **3.123** above function returned value with very high precision and not exact default value (like **3.123000001**)
- Hence to remove this error I added extra work case for floating point case to return same default value as it is provided without giving large precision or wrong answer
- I hope I have answered the question you have asked
- If you do feel some improvement required pls let me know so I could try for better solution
- @vgvassilev

Reply:
Clang’s pretty printer does not print that, so it must be an api that already exists. Maybe that’s controlled by the printing policy.

Reply:
Yes I also think that printing policy is the one that is causing this inappropriate behaviour and searching for a solution to it

Reply:
- @vgvassilev 
- After searching a lot, I feel that even if I change printing policy I will need to check that if argument is float type or not and if so then its precision must be preserved (using modified policy)
- Hence I feel at the end we might need to check for floating type anyways (as we are doing in current code)
- I am just little confused to what are your expectation from code (as current code also works correctly)
- According to me I don't think that we can remove condition of checking float type variable
- So, could you pls explain me your requirements so I could work on it more properly

Reply:
I think I understand now what you are saying. Here is an example which forces clang to print its default arguments: https://godbolt.org/z/vzWxKdsnh 

This seems a deficiency in clang. Perhaps it's worth discussing that at the mailing lists. Clang should print the default argument as written and not as represented.

For the sake of this PR, I think it should be fine to use some rounding mode and add a fixme explaining this implementation detail of clang.

Reply:
- Sir regarding rounding mode I feel that current code gives default value same as written, so won't it be better than using round mode as it will narrow down precision level of arguments to a fixed number
- And even for round mode I will have to change printing police to setPrecision() (for which floating point variable identification will be necessary)
- So should I add round mode or continue with current code + FIXME 
- @vgvassilev